lib/galaxy/webapps/galaxy/fast_app.py

```diff
@@ -72,6 +77,22 @@ async def add_x_frame_options(request: Request, call_next):
             response.headers['X-Frame-Options'] = x_frame_options
             return response
 
+    nginx_x_accel_redirect_base = getattr(gx_app.config, 'nginx_x_accel_redirect_base', None)
+    apache_xsendfile = getattr(gx_app.config, 'apache_xsendfile', None)
+    if nginx_x_accel_redirect_base or apache_xsendfile:
+
+        @app.middleware("http")
+        async def add_send_file_header(request: Request, call_next) -> Response:
+            response = await call_next(request)
+            if not isinstance(response, FileResponse):
+                return response
+            response = cast(FileResponse, response)
+            if nginx_x_accel_redirect_base:
+                response.headers['X-Accel-Redirect'] = nginx_x_accel_redirect_base + response.path
```

Comment:
Will these paths be concatenated to a valid value in all cases? (I don't know)

Reply:
Good question... I replicated 
https://github.com/galaxyproject/galaxy/blob/c132a5031be32bdac7e12f0d66b19ce1b10f022d/lib/galaxy/web/framework/base.py#L492
But maybe the return type `AnyStr@abspath` of `os.path.abspath()` does not behave exactly the same as `Union[str,PathLike[str]]` in `response.path`. I will look into it...

Reply:
Looks like:
```python
val = "/test"
file = FileResponse("/thefile.txt")
result = val + file.path
```
Will concatenate to '/test/thefile.txt' as a string correctly

Reply:
But what if the argument to FileResponse is a relative path? Is that possible? 
Again, I'm not sure, but it seems likely, given this example: https://www.starlette.io/responses/#fileresponse
In that case we'll miss a path delimiter:
```
val = "/test"
file = FileResponse("thefile.txt")
result = val + file.path  # results in /testthefile.txt
```
If (and only if) that's indeed possible, maybe join them as paths? (either using Path or os.path.join?)

Reply:
Good point! I didn't expect it could be a relative path.
Hopefully, https://github.com/davelopez/galaxy/commit/77f0140fc3be580ad8d14b41bb2164da1938c477 will take care of it :)

Reply:
It sure does!